internal/engine/wazevo/wazevoapi/debug_options.go

```diff
@@ -141,9 +135,9 @@ func VerifyOrSetDeterministicCompilationContextValue(ctx context.Context, scope
 	fn := ctx.Value(currentFunctionNameKey{}).(string)
 	key := fn + ": " + scope
 	verifierCtx := ctx.Value(verifierStateContextKey{}).(*verifierState)
-	oldValue, ok := verifierCtx.values[key]
+	oldValue, ok := verifierCtx.values.Load(key)
```

Comment:
If you're going to use a `sync.Map`, please use `LoadOrStore` here to actually make this change "atomic".